mobile/lib/friends/screens/friends_page.dart

```diff
@@ -0,0 +1,211 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:learning_api/api.dart';
+import 'package:mobile/api/ApiClient.dart';
+import 'package:mobile/common/widgets/custom_navigation_bar.dart';
+import 'package:mobile/login_register/extensions/custom_validation_extension.dart';
+import 'package:mobile/profile/widgets/navigation_button.dart';
+import 'package:mobile/quiz/add/widgets/custom_form_field.dart';
+
+import '../../common/helpers/snackbar.dart';
+import '../data/friends_list_data.dart';
+import '../data/get_friends_list.dart';
+
+class FriendsPage extends StatefulWidget {
+  const FriendsPage({super.key});
+
+  @override
+  State<StatefulWidget> createState() => _FriendsPageState();
+}
+
+class _FriendsPageState extends State<FriendsPage> {
+  late Future<FriendsListData> dataFuture;
+  final _formKey = GlobalKey<FormState>();
+
+  String _login = "";
+
+  @override
+  void initState() {
+    super.initState();
+
+    dataFuture = getFriendsData();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final double screenWidth = MediaQuery.of(context).size.width;
+    final double screenHeight = MediaQuery.of(context).size.height;
+
+    return Scaffold(
```

Comment:
Zalecałbym rozbijanie tego na oddzielne widgety, bo wygląda to dość mało czytelnie